63966-Write API requests using Cypress

ioanavavrici opened the pull request:
Enhance API Tests with Comprehensive Error Handling

- Implemented execution for API tests, ensuring all scenarios are tested even if some assertions fail.
- Added error handling for various authentication scenarios, including blank fields, invalid email, and short passwords.
- Included checks for specific error messages and statuses in the response.

ioanavavrici committed:
63966-Write API requests using Cypress
Enhance API Tests with Comprehensive Error Handling

- Implemented execution for API tests, ensuring all scenarios are tested even if some assertions fail.
- Added error handling for various authentication scenarios, including blank fields, invalid email, and short passwords.
- Included checks for specific error messages and statuses in the response.

EmyIacob commented on cypress/e2e/Testing_API/Login Requests.cy.js:
```diff
@@ -0,0 +1,58 @@
+                    'Content-Type': 'application/json'
+                },
+
+            failOnStatusCode: false
```
We should have only one request for positive scenarios only, for testing invalid and valid credentials should have separate tests

EmyIacob commented on cypress/e2e/Testing_API/Signup Requests.cy.js:
```diff
@@ -0,0 +1,151 @@
+    return `${randomString}@example.com`;
+}
+
+const scenarios = [
```
Should be declared in fixture as test data not inside the test file

EmyIacob commented on cypress/e2e/Testing_API/Signup Requests.cy.js:
```diff
@@ -0,0 +1,151 @@
+});
+
+describe('Authentication API Tests', () => {
+    it('Tests various authentication scenarios in parallel', () => {
```
Same, we should have separate tests for positive and negative scenarios, will be hard to debug if is in only one test

EmyIacob commented on cypress/e2e/Testing_API/Signup Requests.cy.js:
```diff
@@ -0,0 +1,151 @@
+                failOnStatusCode: false,
+            }).then((response) => {
+                const jsonData = response.body;
+                cy.log(JSON.stringify(response.body));
```
cy.log or console.log should be used only for debugging.

EmyIacob commented on cypress/e2e/Testing_API/Verify Airports Array and Length.cy.js:
```diff
@@ -0,0 +1,32 @@
+      }
+
+      cy.log('Check number of airports');
+      cy.readFile('cypress/fixtures/credentials.json').then((data) => {
```
Why do you need to read the credentials file?

ioanavavrici committed:
63966-Write API requests using Cypress
- Separated tests for valid credentials, invalid credentials, and missing credentials.

ioanavavrici committed:
I separated tests for valid credentials, invalid credentials, and missing credentials.
